crates/bevy_ecs/src/schedule/executor/multi_threaded.rs

```diff
@@ -523,11 +523,19 @@ impl ExecutorState {
     unsafe fn should_run(
         &mut self,
         system_index: usize,
-        _system: &BoxedSystem,
+        system: &BoxedSystem,
         conditions: &mut Conditions,
         world: UnsafeWorldCell,
     ) -> bool {
+        // Short circuit when system cannot be executed
+        // TODO: move validation before checking system access
```

Comment:
We should also clean this up before merging.

Reply:
This will be pretty intrusive to the existing executors, not sure if it's the right step, I'd like some feedback here if possible

Reply:
Right, so the only benefit is saving some checks right?

Reply:
From discussion around here https://discord.com/channels/691052431525675048/749335865876021248/1286033698327892018
we're fully giving up on short circuiting, so I can remove those todos